mllib/src/main/scala/org/apache/spark/ml/r/GaussianMixtureWrapper.scala

```diff
@@ -91,7 +92,10 @@ private[r] object GaussianMixtureWrapper extends MLReadable[GaussianMixtureWrapp
       .setStages(Array(rFormulaModel, gm))
       .fit(data)
 
-    new GaussianMixtureWrapper(pipeline, dim)
+    val gmm: GaussianMixtureModel = pipeline.stages(1).asInstanceOf[GaussianMixtureModel]
+    val logLikelihood: Double = gmm.summary.logLikelihood
```

Comment:
for this line and above it, why do we need to explicitly give it a type (ie, `Double` or `GaussianMixtureModel`)?